core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -65,24 +65,23 @@ object Action {
   val ConflictCode = 409
 
   /**
-   * Creates a [[CreateAction]].
+   * Creates a [[CreateOrUpdateAction]].
    */
-  def create[T <: ObjectResource](resource: T, editor: ObjectEditor[T])(implicit format: Format[T],
-                                                                        resourceDefinition: ResourceDefinition[T]) =
-    new CreateAction(resource, format, resourceDefinition, editor)
+  def createOrUpdate[T <: ObjectResource](resource: T, editor: ObjectEditor[T])(implicit format: Format[T],
+                                                                                resourceDefinition: ResourceDefinition[T]) =
+    new CreateOrUpdateAction(resource, format, resourceDefinition, editor)
 
   /**
    * Creates a [[DeleteAction]].
    */
   def delete[T <: ObjectResource](resource: T)(implicit resourceDefinition: ResourceDefinition[T]) =
     new DeleteAction(resource, resourceDefinition)
 
-  /**
-   * Creates an [[UpdateAction]].
-   */
-  def update[T <: ObjectResource](resource: T, editor: ObjectEditor[T])(implicit format: Format[T],
-                                                                        resourceDefinition: ResourceDefinition[T]) =
-    new UpdateAction(resource, format, resourceDefinition, editor)
+  def createOrPatch[T <: ObjectResource, O <: Patch](
```

Comment:
The resource is not created immediately in runner actions so for Spark a patch will not be good enough, it will need to be created if it does not exist. This fixes the issue we saw @maasg.

Reply:
I think that `patch`  was required for the Spark operator 0.7 issue with deployments. We should test (prob. later) whether we can do an `update`